dashboard/src/pages/api/run.ts

```diff
@@ -300,11 +311,12 @@ async function run(req: NextApiRequest, res: NextApiResponse) {
         const { dbAlias, sql } = req.body;
         const { username, password } = await getUserAndPassword(tokenInfo, dbAlias);
         const isCust = await isCustomer(username);
-        if (!isCust)
+        if (!isCust) {
           res.status(403).json({
             error: 'User is not a stripe customer',
-            paymentLink: `${config.stripe?.paymentLink}?client_reference_id=${encodeURI(username)}`,
```

Comment:
You may still want to `encodeURI` here because spaces are supposed to be `%20` in a URL.

Reply:
That gets dropped by the regex because it would be invalid and silently dropped by stripe

Reply:
Ahh, got it. I misread the regex and didn't notice the `^` at the beginning of the character class.

Reply:
`uid.replace(/[^a-zA-Z 0-9 _ -]+/g, '');`

This character class includes a space (twice) so we still need to `encodeURI` the output here.